### PR TITLE
feat(router-store): add provideRouterStore function

### LINKED ISSUE #3528: RouterStore: Introduce Standalone API for NgRx RouterStore
### Information

Introduce standalone APIs for using the `@ngrx/router-store` package. This should allow developers to use RouterStore in a "module-free" application.

Based on the discussion in #3399

```ts
provideRouterStore(config)
```

This function will make use of the [ENVIRONMENT_INITIALIZER](https://angular.io/api/core/ENVIRONMENT_INITIALIZER) token to perform any init logic that was previously done in the NgModule constructor.

### Describe any alternatives/workarounds you're currently using

Using `importProvidersFrom` function:

```ts
// main.ts
bootstrapApplication(AppComponent, {
  providers: [importProvidersFrom(StoreRouterConnectingModule.forRoot())]
});
```

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by yharaskrik**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

RouterStore has to be initialized by the `forRoot` method the `StoreRouterConnectingModule` 

Closes #3528

## What is the new behavior?

Can use `provideRouterStore(config)` now to configure the router store

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -497,10 +497,6 @@
               {
                 "command": "nx build-package eslint-plugin"
               },
-              // uncomment when there are eslint-plugin migrations
```
Comment (yharaskrik) [inline angular.json:500]: `yarn format` keeps changing this. Would you like me to revert this so that it's not part of this PR?

**Comment by yharaskrik** [comment]
Ok we should be good to go on this one!

File: modules/router-store/src/init_router_store.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * Initializer for the router-store that ensures the initialization logic is run alongside environment
+ * initialization.
+ */
+export const _initRouterStore: Provider[] = [
+  {
+    provide: ENVIRONMENT_INITIALIZER,
+    multi: true,
+    useValue() {
+      inject(StoreRouterConnectingService);
+    },
```
Comment (brandonroberts) [inline modules/router-store/src/init_router_store.ts:14]: This should be useFactory and return a callback function

```suggestion
    useFactory() {
      return () => {
        inject(StoreRouterConnectingService);
      };
    },
```

File: modules/router-store/src/provide_router_store.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+} from './serializers/base';
+import { _initRouterStore } from './init_router_store';
+
+/**
+ * Provider function to connect the RouterModule to the StoreModule
+ *
+ * Usage:
+ *
+ * ```typescript
+ * @NgModule({
```
Comment (brandonroberts) [inline modules/router-store/src/provide_router_store.ts:26]: This example should be using standalone `bootstrapApplication(AppComponent, { providers: [] })`

While you can use these with NgModules, we want to encourage usage with standalone APIs.

File: angular.json (hunk 1 above)
Comment (brandonroberts) [inline angular.json:500]: Revert this change

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: angular.json (hunk 1 above)
Comment (yharaskrik) [reply angular.json:500]: Sounds good. Not sure why yarn format is doing that but every time formatting is run this gets removed.

File: modules/router-store/src/provide_router_store.ts (hunk 3 above)
Comment (yharaskrik) [reply modules/router-store/src/provide_router_store.ts:26]: Would you like me to assume there is a `provideStore` and such in the example? or just add `provideRouterStore` to the example?

File: modules/router-store/src/init_router_store.ts (hunk 2 above)
Comment (yharaskrik) [reply modules/router-store/src/init_router_store.ts]: Does the env initializer token need it to be a factory? Cause all of the tests passed so I think useValue worked?

File: modules/router-store/src/init_router_store.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    provide: ENVIRONMENT_INITIALIZER,
+    multi: true,
+    useFactory() {
+      return () => {
+        inject(StoreRouterConnectingService);
+      };
+    },
+  },
+  StoreRouterConnectingService,
+];
```
Comment (markostanimirovic) [inline modules/router-store/src/init_router_store.ts:19]: We can delete this file and move `ENVIRONMENT_INITIALIZER` to `provideRouterStore`.

File: modules/router-store/src/provide_router_store.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+      provide: RouterStateSerializer,
+      useClass: config.serializer
+        ? config.serializer
+        : config.routerState === RouterState.Full
+        ? FullRouterStateSerializer
+        : MinimalRouterStateSerializer,
+    },
+    _initRouterStore,
+  ];
+}
```
Comment (markostanimirovic) [inline modules/router-store/src/provide_router_store.ts:53]: We decided to use `EnvironmentProviders` as the return type of standalone functions to throw a compilation error if they're used in the wrong place (e.g. `Component/Directive.providers`).

```suggestion
export function provideRouterStore<
  T extends BaseRouterStoreState = SerializedRouterStateSnapshot
  // import `EnvironmentProviders` from `@ngrx/store` package
>(config: StoreRouterConfig<T> = {}): EnvironmentProviders {
  return {
    ɵproviders: [
      { provide: _ROUTER_CONFIG, useValue: config },
      {
        provide: ROUTER_CONFIG,
        useFactory: _createRouterConfig,
        deps: [_ROUTER_CONFIG],
      },
      {
        provide: RouterStateSerializer,
        useClass: config.serializer
          ? config.serializer
          : config.routerState === RouterState.Full
          ? FullRouterStateSerializer
          : MinimalRouterStateSerializer,
      },
      StoreRouterConnectingService,
      {
        provide: ENVIRONMENT_INITIALIZER,
        multi: true,
        useFactory() {
          return () => inject(StoreRouterConnectingService);
        },
      },
    ],
  };
}
```

File: modules/router-store/src/router_store_module.ts [hunk 6]
```diff
@@ -158,231 +48,14 @@ enum RouterTrigger {
...
-        {
-          provide: RouterStateSerializer,
-          useClass: config.serializer
-            ? config.serializer
-            : config.routerState === RouterState.Full
-            ? FullRouterStateSerializer
-            : MinimalRouterStateSerializer,
-        },
-      ],
+      providers: provideRouterStore(config),
```
Comment (markostanimirovic) [inline modules/router-store/src/router_store_module.ts:58]:
```suggestion
      providers: [...provideRouterStore(config).ɵproviders],
```

**Review by markostanimirovic — CHANGES_REQUESTED** [review]

File: modules/router-store/src/provide_router_store.ts [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  BaseRouterStoreState,
+  RouterStateSerializer,
+} from './serializers/base';
+import { StoreRouterConnectingService } from './store_router_connecting.service';
+import { EnvironmentProviders } from '@ngrx/store';
+
+/**
+ * Provider function to connect the RouterModule to the StoreModule
+ *
+ * Usage:
```
Comment (markostanimirovic) [inline modules/router-store/src/provide_router_store.ts:24]:
```suggestion
 * Connects the Angular Router to the Store.
 *
 * @usageNotes
```

**Review by markostanimirovic — APPROVED** [review]
LGTM 👍  Great work Jay!

One small suggestion regarding API docs:

**Review by brandonroberts — APPROVED** [review]
